### ISSUE TITLE
Add 2FA

### ISSUE THREAD

**Issue by foozmeat** [jointakahe/takahe#390]
It would be nice to be able to enforce it server-wide.

**Comment by andrewgodwin** [comment]
Agreed - this has been on my mental list for a while. I think my current position is TOTP and Webauthn only, SMS two factor would need more provider integration and also isn't amazing anyway.

**Comment by foozmeat** [comment]
+1 to Webauthn

**Comment by pauloxnet** [comment]
There's a Django package for that: Kagi
_Kagi provides support for FIDO WebAuthn security keys and TOTP tokens in Django._ 
https://github.com/justinmayer/kagi

**Comment by tabletcorry** [comment]
+1 and also consider supporting Passkey/WebauthN _only_ login, where passwords aren't collected or allowed. I would love to have a switch where passwords aren't a concern anymore.

**Comment by andrewgodwin** [comment]
Good idea @tabletcorry, though we'd need a new reset flow for that situation so you could get back in.

**Comment by tabletcorry** [comment]
Nod, I think emails would still be attached to the user so the "password reset flow" would become a "add a new passkey" flow when appropriate. Hard to replace email as a core root of identity...
